conformance/utils/suite/features.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import "k8s.io/apimachinery/pkg/util/sets"
+
+// -----------------------------------------------------------------------------
+// Features - Types
+// -----------------------------------------------------------------------------
+
+// SupportedFeature allows opting in to additional conformance tests at an
+// individual feature granularity.
+type SupportedFeature string
+
+// -----------------------------------------------------------------------------
+// Features - Standard (Core)
+// -----------------------------------------------------------------------------
+
+const (
+	// This option indicates support for ReferenceGrant (core conformance).
+	// Opting out of this requires an implementation to have clearly implemented
+	// and documented equivalent safeguards.
+	SupportReferenceGrant SupportedFeature = "ReferenceGrant"
+)
+
+// StandardCoreFeatures are the features that are required to be conformant with
+// the Core API features (e.g. GatewayClass, Gateway, e.t.c.).
+//
+// TODO: rename this from "Standard" as that might be confusing with our
+// "Standard" release channel.
+var StandardCoreFeatures = sets.New(
+	SupportReferenceGrant,
+)
```

Comment:
Maybe we should just remove this for the time being and add it back in when we have a clearer picture of what it would look like? Just have a separate set with ReferenceGrant doesn't seem very helpful.

Reply:
You made another comment about this below with more prescription about the changes you'd actually like to see, I'll consider this thread resolved in favor of continuing the conversation there.